dsperse/src/run/runner.py

```
@@ -327,36 +332,262 @@ def _generate_run_metadata(self, format: str = "dirs"):
+                    if missing:
+                        raise ValueError(f"Missing input tensors for {current_slice_id}: {missing}")
+                    extra_tensors = {name: tensor_cache[name] for name in filtered_inputs}
+                    ok, result, exec_info = self._run_slice_multi_input(info, out_file, slice_dir, extra_tensors)
+                else:
+                    ok, result, exec_info = RunnerUtils.execute_slice(self, nodes[current_slice_id], info, in_file, out_file, slice_dir)
+
+                if ok and result:
+                    if isinstance(result, dict) and 'output_tensors' in result:
+                        for oname, tensor in result['output_tensors'].items():
+                            tensor_cache[oname] = tensor
+                        first_output = list(result['output_tensors'].values())[0] if result['output_tensors'] else None
```

Ruff (0.14.11)

Prefer `next(iter(result['output_tensors'].values()))` over single element slice

Replace with `next(iter(result['output_tensors'].values()))`

(RUF015)